gem/optimise.py

```diff
@@ -1056,50 +1056,176 @@ class Kernel_Optimiser(object):
             factors depend on indices (typically quadrature indices for reduction)
             other than argument indices
     """
-    def __init__(self, node, arg_ind):
+    def __init__(self, node, arg_ind, rep=None):
         """
         Constructor
         :param node: gem node
         :param arg_ind: tuples of tuples of argument (linear) indices
         """
         self.node = node
         self.arg_ind = arg_ind
+        # TODO: make these two properties of the object
+        self.arg_ind_flat = tuple([i for id in self.arg_ind for i in id])
+        self.arg_ind_set = set(self.arg_ind_flat)
```

Comment:
Why do you need three versions of argument indices? You don't even use the original version... (`arg_ind`)